Sprint 1/step 1

piladik opened the pull request.

piladik committed:
test commit to sprint-1/step-1

piladik committed:
added header

piladik committed:
draft markup

piladik committed:
added scrollbars

piladik committed:
added PropTypes validation for BurgerIngredients.js and BurgerConstructor.js

ArslanDev666 commented on src/components/header/Header.js:
```diff
@@ -0,0 +1,28 @@
+                    <Logo />
+                </div>
+                <div className={navStyles.nav_box}>
+                    <HeaderAccount text="Личный кабинет"/>
```
**Надо исправить:**

Похожее замечание написал выше, это опять таки всего лишь ссылка.

ArslanDev666 commented on src/components/main/burger-constructor/BurgerConstructor.js:
```diff
@@ -0,0 +1,90 @@
+import BurgerConstructorConfirm from "./BurgerConstructorConfirm";
+
+function BurgerConstructor(props) {
+    const buns = props.ingredients.filter(el => el.type === "bun");
+    const sauces = props.ingredients.filter(el => el.type === "sauce");
+    const mains = props.ingredients.filter(el => el.type === "main");
```
**Надо исправить:**

Тут лучше сделать filter следующим образом:

```js
const bun = data.find(el => el.type === "bun");
const ingredients = data.filter(el => el.type !== "bun");
```

И использовать сразу просто bun. А также использовать map, чтобы вывести остальные ингредиенты.

ArslanDev666 commented on src/components/main/burger-constructor/BurgerConstructor.js:
```diff
@@ -0,0 +1,90 @@
+                    <ConstructorElement
+                        type="top"
+                        isLocked={true}
+                        text={buns[0].name}
```
**Надо исправить:**

здесь и ниже, у булок по макету есть приписка - `(верх)` и `(низ)`.

ArslanDev666 commented on src/components/main/burger-constructor/BurgerConstructor.js:
```diff
@@ -0,0 +1,90 @@
+    )
+}
+
+BurgerConstructor.propTypes = PropTypes.shape({
+    _id: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    type: PropTypes.oneOf(["bun, main, sauce"]).isRequired,
+    proteins: PropTypes.number.isRequired,
+    fat: PropTypes.number.isRequired,
+    carbohydrates: PropTypes.number.isRequired,
+    calories: PropTypes.number.isRequired,
+    price: PropTypes.number.isRequired,
+    image: PropTypes.string,
+    image_mobile: PropTypes.string,
+    image_large: PropTypes.string,
+    __v: PropTypes.number
+})
```
**Надо исправить:**

Вы правильно описали структуру ингредиента, но вы неправильно описали пропсы для самого компонента.

```js
ingredients: PropTypes.arrayOf(ingredientTypes.isRequired).isRequired;
```

должно быть что-то типа такого.

Следует вынести структуру в отдельный файл (можно создать файл prop-types.js в папке utils) и переиспользовать описание пропсов ингредиента, а не дублировать.

ArslanDev666 commented on src/components/main/burger-constructor/BurgerConstructorConfirm.js:
```diff
@@ -0,0 +1,18 @@
+        <div className={`mt-10 ${styles.confirm_container}`}>
+            <div className={`mr-10 ${styles.total_box}`}>
+                <p className="text text_type_digits-medium">610</p>
+                <CurrencyIcon type="primary" className={`${styles.icon}`} />
```
**Можно лучше:**

ClassName лучше указать вот так:

```js
className={styles.icon}
```

Мы ведь ничего не вычисляем, а значит шаблонные строки нам не нужны.

ArslanDev666 commented on src/components/main/burger-ingredients/BurgerIngredients.js:
```diff
@@ -0,0 +1,59 @@
+import IngredientsCard from "./IngredientsCard.js";
+
+function BurgerIngredients(props) {
+    const buns = props.ingredients.filter(el => el.type === "bun");
+    const sauces = props.ingredients.filter(el => el.type === "sauce");
+    const mains = props.ingredients.filter(el => el.type === "main");
```
**Вы молодец!**

Вы правильно разделили ингредиенты.

ArslanDev666 commented on src/components/main/burger-ingredients/BurgerIngredients.js:
```diff
@@ -0,0 +1,59 @@
+            <IngredientsMainHeading />
+            <IngredientsTab />
+            <div className={`${styles.scrollable_box}`}>
+                <IngredientsSecondaryHeading text="Булки"/>
```
**Надо исправить:**

Здесь и далее.
Не стоит выносить такой маленький кусочек кода в отдельный компонент. 

Вы можете заметить, что у каждой категории ингредиентов дублируется верстка, лучше её вынести отдельный компонент ingredients-category и переиспользовать его.

ArslanDev666 commented on src/components/main/burger-ingredients/BurgerIngredients.js:
```diff
@@ -0,0 +1,59 @@
+    )
+}
+
+BurgerIngredients.propTypes = PropTypes.shape({
+    _id: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    type: PropTypes.oneOf(["bun, main, sauce"]).isRequired,
+    proteins: PropTypes.number.isRequired,
+    fat: PropTypes.number.isRequired,
+    carbohydrates: PropTypes.number.isRequired,
+    calories: PropTypes.number.isRequired,
+    price: PropTypes.number.isRequired,
+    image: PropTypes.string,
+    image_mobile: PropTypes.string,
+    image_large: PropTypes.string,
+    __v: PropTypes.number
+})
```
**Надо исправить:**

Аналогичное замечание было в `burger-constructor`.

ArslanDev666 commented on src/components/main/burger-ingredients/IngredientsCard.js:
```diff
@@ -0,0 +1,41 @@
+import styles from "./BurgerIngredients.module.css";
+
+function IngredientsCard(props) {
+    if ( props.ingredient.name == "Краторная булка N-200i" ||
+         props.ingredient.name == "Говяжий метеорит (отбивная)" ||
+         props.ingredient.name == "Биокотлета из марсианской Магнолии" ||
+         props.ingredient.name == "Соус Spicy-X" ||
+         props.ingredient.name == "Соус традиционный галактический"
```
**Надо исправить:** 

Давайте пока что уйдём от условий, и будем по умолчанию всем выводить count, сейчас это не так важно))

Старайтесь применять всегда строгое равенство.

ArslanDev666 commented on src/components/main/burger-ingredients/IngredientsCard.js:
```diff
@@ -0,0 +1,41 @@
+            return (
+                <div className={`mt-6 mb-10 ml-4 ${styles.card}`}>
+                    <Counter count={1} size="default" extraClass="m-1" />
+                    <img className={`ml-4 mr-4`} src={props.ingredient.image}></img>
```
**Надо исправить:** 

У элемента `img` должен быть атрибут `alt`.

Элемент `img`, можно записать вот так. 

```jsx
<img className="ml-4 mr-4" alt={props.ingredient.name} src={props.ingredient.image} />
```

ArslanDev666 commented on src/components/main/burger-ingredients/IngredientsCard.js:
```diff
@@ -0,0 +1,41 @@
+import { CurrencyIcon, Counter } from '@ya.praktikum/react-developer-burger-ui-components';
+import styles from "./BurgerIngredients.module.css";
+
+function IngredientsCard(props) {
```
**Надо исправить:**

Для этого компонента так же нужно описать PropTypes.

ArslanDev666 commented on src/components/main/burger-ingredients/BurgerIngredients.js:
```diff
@@ -0,0 +1,59 @@
+    const mains = props.ingredients.filter(el => el.type === "main");
+    return (
+        <section className={`mt-10 ${styles.ingredients_section}`}>
+            <IngredientsMainHeading />
```
**Надо исправить:**

Не стоит выносить в отдельный компонент 1 элемент. Лучше сразу всё написать в этом файле.

ArslanDev666 commented on src/App.js:
```diff
@@ -0,0 +1,23 @@
+    const ingredients = data;
+    return (
+        <div className="App text text_type_main-default">
+        <Header test="This prop has been passed successfully"/>
```
**Надо исправить:**

вроде бы у хедера нет пропсов.

ArslanDev666 commented on src/App.js:
```diff
@@ -0,0 +1,23 @@
+import data from "./utils/data.js";
+
+function App() {
+    const ingredients = data;
```
**Можно лучше:**

Не совсем понимаю зачем тут это)) почему бы сразу не использовать `data`.

ArslanDev666 commented on src/App.tsx:
```diff
@@ -2,23 +2,12 @@ import React from 'react';
 
+import Header from './components/header/Header.js';
+
 function App() {
```
**Надо исправить:**

Разберитесь какой компонент `App` лучше оставить, а то у вас их целых 2)).

ArslanMustafin reviewed (changes requested):
## Code Review

Здравствуйте!

Отличная работа.

Приложение начинает обрастать функционалом и уже постепенно проглядывается картина готового продукта.

### **Вы молодец!**

- Декомпозиция файлов по директориям соответствует описанию;
- Сборка и запуск проекта выполняются без ошибок;
- Сверстаны всё необходимые элементы;
### **Но есть несколько замечаний, которые нужно исправить:**

- Нужно очистить корневую директорию от лишних файлов, которые остались при инициализации create-react-app.

  - App.test.tsx
  - logo.svg

- В компоненте App-header требуется поправить семантику, икона тоже должна быть обёрнута в ссылку. Также, по макету должна активна только первая ссылка.

- В директории `components` должен быть компонент App. Компонент `App`, он также должен быть в папке (как и другие компоненты). Как правило, это основной компонент в React, который выступает в качестве контейнера для всех остальных компонентов;

- В burger-constructor у булок, должна быть приписка - `(верх)` и `(низ)`;

- Лучше поменять расширение у файлов с `.js` на `.jsx`. Это касается только тех файлов, где содержится `jsx`.

- Лучше не создавать лишней вложенности, в виде подпапки `main`, это может усложнить дальнейшее расширение проекта, а также его рефакторинг.


- Не стоит писать стили инлайново, в будущем это может затруднить поиск ошибок, и перебить стили заданных классов. Лучше дать какой-то класс, и там указать уже нужные вам стили.


- Не используйте там где простые строки, шаблонные строки. 

  Пример: 
  ```jsx
    <img className={`ml-4 mr-4`} src={props.ingredient.image}>

    // Лучше написать вот так

    <img className="ml-4 mr-4" src={props.ingredient.image}>
  ```

### **Сделает код лучше:**

- Перед отправкой работы убедитесь в том, что весь неиспользуемый код удален.

- Стоит обращать внимание на предупреждения в терминале, в котором запущен проект. Это хорошие рекомендации, которых следовало бы придерживаться. В противном случае, лучше отключить линтер у строки (куска кода) eslint-disable. Так же добавить комментарий (TODO), объясняющий причину блокировки линтера в данном месте. Суть в том, что мы стараемся писать более чистый код, а если видим, что порой выскакивают предупреждения того же линтера, то оставляем комментарий, объясняющий причину. В коммерческой разработке, когда есть целая команда разработчиков, работающих над одним проектом, эта рекомендация послужит правилом хорошего тона. Неплохая статься по поводу локального отключения линтера: https://learn.coderslang.com/ru/0023-eslint-disable-for-specific-lines-files-and-folders

- В некоторых местах нарушено форматирование кода, проблемы с отступами и в некоторых местах слишком длинные строки, что затрудняет чтение кода. Об оформлении кода можно почитать здесь https://learn.javascript.ru/coding-style. Постарайтесь использовать автоматическое форматирование, это сильно экономит время. Одно из наиболее популярных дополнений для форматирования кода - Prettier (https://prettier.io/)

### **_Подробно эти и более мелкие замечания, отмечены в коде._**

#### **Хочу напомнить, что работа может быть принята, только после исправления всех "Надо исправить". Перед отправкой на ревью, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка F12).**

piladik committed:
1. Удалил
-App.test.tsx
-logo.svg
2. Исправил в соответствии в рекоменданцией - Лучше не создавать лишней вложенности, в виде подпапки main, это может усложнить дальнейшее расширение проекта, а также его рефакторинг.

piladik committed:
Исправил все недочеты с тегом 'Надо исправить'

piladik committed:
Установил prettier и отформатировал все файлы

piladik committed:
Изменил название файла /header/Header.module.css на /header/header.module.css так как css модуль не импортировался и проект не запускался

ArslanDev666 commented on src/components/burger-constructor/burger-constructor.jsx:
```diff
@@ -0,0 +1,67 @@
+import { ingredientPropTypes } from "../../utils/prop-types";
+
+function BurgerConstructor(props) {
+  const bun = props.ingredients.find((el) => el.type === "bun");
+  const ingredients = props.ingredients.filter((el) => el.type !== "bun");
```
**Вы молодец!**

- вы правильно разделили булку, и остальные ингредиенты.

ArslanDev666 commented on src/components/burger-constructor/burger-constructor.jsx:
```diff
@@ -0,0 +1,67 @@
+}
+
+BurgerConstructor.propTypes = {
+  ingredients: PropTypes.arrayOf(ingredientPropTypes.isRequired),
```
**Надо исправить:**

Необходимо указывать .isRequired как у элемента массива, так и у самого массива в propTypes.

Пример как правильно:

```js
data: PropTypes.arrayOf(ingredientTypes.isRequired).isRequired;
```

ArslanDev666 commented on src/components/burger-ingredients/burger-ingredients.jsx:
```diff
@@ -0,0 +1,57 @@
+}
+
+BurgerIngredients.propTypes = {
+  ingredients: PropTypes.arrayOf(ingredientPropTypes.isRequired),
```
**Надо исправить:**

Необходимо указывать .isRequired как у элемента массива, так и у самого массива в propTypes.

Пример как правильно:

```js
data: PropTypes.arrayOf(ingredientTypes.isRequired).isRequired;
```

ArslanDev666 commented on src/components/burger-ingredients/ingredients-card.jsx:
```diff
@@ -0,0 +1,34 @@
+}
+
+IngredientsCard.propTypes = {
+  ingredient: ingredientPropTypes,
```
**Надо исправить:**

Не забывайте указывать `isRequired`, без этого пропсы не проверяются.

ArslanMustafin reviewed (approved):
## Code Review

Здравствуйте!

Отличная работа.

### **Вы молодец! Можете приступать к финальному этапу задания**

- Практически все замечания с прошлого ревью были исправлены.
  
### **Но есть несколько замечаний, которые нужно исправить (Исправления должны быть уже в финальном этапе задания):**

- У вас ещё осталась проблема с регистром файла, https://yadi.sk/d/BrAxxnGcPBGBiA; У гита есть заморочки с регистром у файлов, поэтому прочтите например https://noteskeeper.ru/heritage/672/, чтобы правильно пофиксить проблему;

- В компоненте App-header требуется поправить семантику, **Икона** тоже должна быть обёрнута в ссылку.

- В burger-constructor у булок, должна быть приписка - `(верх)` и `(низ)`;

- Лучше поменять расширение у файлов с `.js` на `.jsx`. Это касается только тех файлов, где содержится `jsx`.
  - Файл с компонентом `App`, тоже должен иметь расширение `.jsx`.

- Лучше удалить не используемые компоненты: 
  - ingredients-main-heading;
  - ingredients-secondary-heading;
  - App.js (в корне проекта);
  - logo.svg (в корне проекта);

- Стоит переименовать: 
  - `header` в `app-header` (папку и компонент), чтобы бы всё совпадало с чек-листом));
  - файл (основной и стилей) компонента App, сделать со строчной буквы, чтобы было единобразие.

piladik committed:
deleted header.jsx

piladik committed:
added header.jsx

piladik committed:
deleted header.module.css

piladik committed:
added header.module.css

piladik committed:
test

piladik committed:
Merge branch 'main' of https://github.com/piladik/react-burger

piladik committed:
Resolved merge conflict
